tests/models/test_utils.py

```diff
@@ -321,36 +321,94 @@ def test_enforce_property_with_errors():
         )
 
 
-def test_enforce_array():
-    data = ["some_sentence1", "some_sentence2"]
-    arr = Array(DataType.string)
-    assert _enforce_array(data, arr) == data
+@pytest.mark.parametrize(
+    ("data", "schema"),
+    [
+        # 1. Simply array
+        (["some_sentence1", "some_sentence2"], Array(DataType.string)),
+        # 2. Numpy
+        (np.array(["some_sentence1", "some_sentence2"]), Array(DataType.string)),
+        # 3. Array of Object
+        (
+            [
+                {"a": "some_sentence1", "b": "some_sentence2"},
+                {"a": "some_sentence3", "c": ["some_sentence4", "some_sentence5"]},
+            ],
+            Array(
+                Object(
+                    [
+                        Property("a", DataType.string),
+                        Property("b", DataType.string, required=False),
+                        Property("c", Array(DataType.string), required=False),
+                    ]
+                )
+            ),
+        ),
+        # 4. Nested array
+        (
+            [
+                [["a", "b"], ["c", "d"]],
+                [["e", "f", "g"], ["h"]],
+                [["g"]],
+            ],
+            Array(Array(Array(DataType.string))),
+        ),
```

Comment:
@serena-ruan Is there a limit for nesting depth?

Reply:
Nope

Reply:
Should we add one?

Reply:
Practically I don't think someone inputs 10x nested list, but good to have such guardrail. Probably we can put some handy limit like ~5 and raise it once some user hits that (and report)?

Reply:
How about 2? We can increase it if there is a feature request.

Reply:
That's fine too. We have TensorSpec for use cases with high dimensional array like deep learning right?